### PR TITLE
Add swapBOBAForETHMetaTransaction

### PR DESCRIPTION
**Pull request by boyuan-chen**
- [x] Update `Boba_GasPriceOracle` contract
- [x] Add unit tests
- [x] Add integration tests
- [x] Update API endpoint
- [x] Update graph node

### CODE REVIEW

**Review by CAPtheorem — APPROVED** [review]

File: packages/contracts/contracts/L2/predeploys/Boba_GasPriceOracle.sol [hunk 1]
```diff
@@ -139,7 +163,7 @@ contract Boba_GasPriceOracle {
      * using the Meta Transaction
      * NOTE: Only works for the mainnet and local testnet
      */
-    function useBobaAsFeeTokenMetaTransaction(
+    function swapBOBAForETHMetaTransaction(
```
Comment (souradeep-das) [inline packages/contracts/contracts/L2/predeploys/Boba_GasPriceOracle.sol:171]: i personally think the previous name for this method was better. is the new name kinda confusing?

File: packages/contracts/test/contracts/L2/predeploys/Boba_GasPriceOracle.spec.ts [hunk 2]
```diff
@@ -398,6 +398,100 @@ describe('Boba_GasPriceOracle', () => {
...
+          receivedETHAmount
+        )
+      )
+        .to.emit(Boba_GasPriceOracle, 'UpdateReceivedETHAmount')
+        .withArgs(await signer1.getAddress(), receivedETHAmount)
+    })
+  })
+
+  describe('get receivedETHAmount', () => {
+    it('should revert if caller is not owne', async () => {
```
Comment (souradeep-das) [inline packages/contracts/test/contracts/L2/predeploys/Boba_GasPriceOracle.spec.ts]:
```suggestion
    it('should revert if caller is not owner', async () => {
```

**Review by souradeep-das — APPROVED** [review]
🚀

File: packages/contracts/contracts/L2/predeploys/Boba_GasPriceOracle.sol (hunk 1 above)
Comment (boyuan-chen) [reply packages/contracts/contracts/L2/predeploys/Boba_GasPriceOracle.sol:171]: We provide a small swap for users that don't have enough ETH when they switch fee token to Boba.

Comment (souradeep-das) [reply packages/contracts/contracts/L2/predeploys/Boba_GasPriceOracle.sol:171]: but is this is also the useBobaAsFeeToken meta transaction?

Comment (boyuan-chen) [reply packages/contracts/contracts/L2/predeploys/Boba_GasPriceOracle.sol:171]: No, we don't need another meta transaction for switching the fee token. Even users switch to Boba as the fee token, they have to have some ETH in their wallets. MetaMask checks the user's balance and blocks the tx if there is no ETH in their wallet.

The logic for normal users that only bridge BOBA to L2 is that they use this meta transaction to swap some ETH and then change fee token to BOBA using the normal tx. They have some ETH in their wallet and also they use the BOBA as the fee token.

Comment (souradeep-das) [reply packages/contracts/contracts/L2/predeploys/Boba_GasPriceOracle.sol:171]: got it, alright!

**Pull request merged**